core/src/main/scala/org/http4s/headers/Strict-Transport-Security.scala

```diff
@@ -1,7 +1,25 @@
 package org.http4s
 package headers
 
+import scala.concurrent.duration.FiniteDuration
+import org.http4s.parser.HttpHeaderParser
+import org.http4s.util.Writer
+
 /**
  * Defined by http://tools.ietf.org/html/rfc6797
  */
-object `Strict-Transport-Security` extends HeaderKey.Default
+object `Strict-Transport-Security` extends HeaderKey.Internal[`Strict-Transport-Security`] with HeaderKey.Singleton {
+  def parse(s: String): ParseResult[`Strict-Transport-Security`] =
+    HttpHeaderParser.STRICT_TRANSPORT_SECURITY(s)
+}
+
+case class `Strict-Transport-Security`(maxAge: FiniteDuration, includeSubDomains: Boolean = true, preload: Boolean = false) extends Header.Parsed {
```

Comment:
Yes, very unfortunate that `Duration` misbehaves. Wonder if there is a clean proposal we could come up with for scalac?